examples/diffusers/quantization/quantize.py

```diff
@@ -219,6 +219,104 @@ def parse_args() -> argparse.Namespace:
 
     return parser.parse_args()
 
+def quantize_diffusion_model(pipe: DiffusionPipeline,
+                             backbone: torch.nn.Module,
+                             model_name: str = "flux-dev",
+                             precision: str = "fp8",
+                             quant_level: float = 3.0,
+                             calib_size: int = 128, 
+                             batch_size: int = 2, 
+                             calibration_source: str = "./calib/calib_prompts.txt",
+                             collect_method : str = "default",
+                             quant_algo : str = "smoothquant", 
+                             percentile: float = 1.0, 
+                             num_denoising_steps: int = 30, 
+                             trt_high_precision_dtype: str = "Half",
+                             alpha: float = 1.0,
+                             lowrank: int = 32) -> torch.nn.Module:
+    """
+    Quantize a diffusion model using the given quant_config.
+    """
+    if quant_level == 4.0 and format == "int8":
+        raise ValueError("Level 4 quantization is only supported for fp8, not int8.")
+    
+    # Validate model_name exists in one of the supported models
+    if model_name not in ["sd3-medium", "flux-dev", "flux-schnell"]:
+        raise ValueError(f"model_name must be one of ["sd3-medium", "flux-dev", "flux-schnell"], got {model_name}")
+
+    # Adjust calibration steps to be number of batches
+    calib_size = calib_size // batch_size
+
+    calibration_prompts = load_calib_prompts(
+            batch_size,
+            calibration_source,
+        )
+
+    # Build quant_config based on format
+    if precision == "int8":
+        if collect_method == "default":
+                raise ValueError(
+                    "You must specify an explicit --collect-method (e.g., 'global_min') for int8."
+                )
+        if quant_algo != "smoothquant":
+            raise ValueError(
+                "INT8 quantization only works well when combined with SmoothQuant;"
+                "otherwise, it will produce very poor quality results."
+            )
+        quant_config = get_int8_config(
```

Comment:
Also wouldnt it make sense to just have the user provide the quant config instead of enumerating all the options?